src/market/agreement/agreement-event.ts

```diff
@@ -0,0 +1,26 @@
+import { Agreement } from "./agreement";
+
+export type AgreementEvent =
+  | {
```

Comment:
name these types explicitly, so that they can be referenced in the Market events list,